internal/release/github.go

```diff
@@ -0,0 +1,171 @@
+	var httpClient *http.Client
+	if ghToken := os.Getenv("GITHUB_TOKEN"); ghToken != "" {
+		log.Printf("creating authenticated client with GITHUB_TOKEN")
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: ghToken},
+		)
+		httpClient = oauth2.NewClient(ctx, ts)
+	} else {
+		httpClient = retryablehttp.NewClient().StandardClient()
+	}
+	return &Client{
+		GitHub: github.NewClient(httpClient),
+	}
+}
+
+// CreateRelease creates a new GitHub release under the owner and repo.
+func (c *Client) CreateRelease(ctx context.Context, owner GithubOwner, repo GithubRepo, release *github.RepositoryRelease) (*github.RepositoryRelease, error) {
+	repositoryRelease, _, err := c.GitHub.Repositories.CreateRelease(ctx, string(owner), string(repo), release)
+	if err != nil {
+		return nil, err
+	}
+	return repositoryRelease, nil
+}
+
+// EditRelease performs an update of editable properties of a release (i.e. marking it not as a draft).
+func (c *Client) EditRelease(ctx context.Context, owner GithubOwner, repo GithubRepo, releaseID int64, releaseChanges *github.RepositoryRelease) (*github.RepositoryRelease, error) {
+	release, _, err := c.GitHub.Repositories.EditRelease(ctx, string(owner), string(repo), releaseID, releaseChanges)
+	if err != nil {
+		return nil, err
+	}
+	return release, nil
+}
+
+// UploadReleaseAsset uploads the specified file to the release.
+func (c *Client) UploadReleaseAsset(ctx context.Context, owner GithubOwner, repo GithubRepo, releaseID int64, filename string) error {
+	f, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	_, _, err = c.GitHub.Repositories.UploadReleaseAsset(ctx, string(owner), string(repo), releaseID, &github.UploadOptions{
+		Name: filepath.Base(filename),
+	}, f)
+	return err
+}
+
+// GetLatestRelease returns information about the latest GitHub release for the given org and repo (i.e. 'bufbuild', 'plugins').
+// If no release is found, returns ErrReleaseNotFound.
+func (c *Client) GetLatestRelease(ctx context.Context, owner GithubOwner, repo GithubRepo) (*github.RepositoryRelease, error) {
+	repositoryRelease, resp, err := c.GitHub.Repositories.GetLatestRelease(ctx, string(owner), string(repo))
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			// no latest release
+			return nil, ErrNotFound
+		}
+		return nil, err
+	}
+	return repositoryRelease, nil
+}
+
+// GetReleaseByTag returns information about a given release (by tag name).
+func (c *Client) GetReleaseByTag(ctx context.Context, owner GithubOwner, repo GithubRepo, tag string) (*github.RepositoryRelease, error) {
+	repositoryRelease, resp, err := c.GitHub.Repositories.GetReleaseByTag(ctx, string(owner), string(repo), tag)
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			// no latest release
+			return nil, ErrNotFound
+		}
+		return nil, err
+	}
+	return repositoryRelease, nil
+}
+
+func getOrgRepoFromReleaseURL(url string) (string, string, error) {
```

Comment:
Maybe a personal nit, but it makes it a bit easier to reason about if there's an example comment of the input.

Reply:
Good call - I'll add an example.